py/desispec/scripts/submit_night.py

```diff
@@ -490,7 +490,7 @@ def submit_night(night, proc_obstypes=None, z_submit_types=None, queue='realtime
         sys.stdout.flush()
         sys.stderr.flush()
 
-    if tableng > 0:
+    if tableng >= 0:
```

Comment:
This is equivalent to removing the `if` statement, which is there to protect against cases where there genuinely is no data on a given night. 

I'd recommend modifying the `if` statement instead:

`if tableng > 0:`   -->  `if tableng > 0 or (use_tilenight and len(sciences)>0):`

An alternate solution would be to ensure all calling functions are robust to empty arguments (empty `sciences`  passed to `checkfor_and_submit_joint_job`, no entries in `ptable` passed to `update_from_queue`, etc.

The third option would be to throw an error there if there is no good data rather than having the code crash. I slightly disfavor this, but am open to discussion.

Personally, I think the first option is the least invasive until we officially retire the non tilenight code in the script rewrite.

Reply:
1 done